frontend/src/__tests__/cypress/cypress/utils/oc_commands/project.ts

```diff
@@ -71,3 +71,43 @@ export const addUserToProject = (
     return result;
   });
 };
+
+/**
+ * Verify if an OpenShift Project exists
+ *
+ * @param projectName OpenShift Project name to verify (in the format like 'cypress-test-project')
+ * @returns A Cypress chainable boolean indicating whether the project exists (true) or not (false)
+ */
+export const verifyOpenShiftProjectExists = (projectName: string): Cypress.Chainable<boolean> => {
+  if (!projectName || typeof projectName !== 'string') {
+    cy.log(`ERROR: Invalid project name provided: ${projectName}`);
+    throw new Error(`Invalid project name: ${projectName}`);
+  }
+
+  const checkCommand = `oc get project "${projectName}" -o name`;
+
+  cy.log(`Executing command: ${checkCommand}`);
+
+  return cy.exec(checkCommand, { failOnNonZeroExit: false }).then((result: Cypress.Exec) => {
+    cy.log(`Command result: ${JSON.stringify(result)}`);
+
+    if (result.code !== 0 && result.code !== 1) {
+      // Code 1 is expected when the project doesn't exist, so we only throw for other non-zero codes
+      cy.log(`ERROR: Command execution failed
```

Comment:
I think we should rather use a common function as you do with `cy.step` , for example, add this to utils or helpers:
```
// utils.js or helpers.js
function logAndThrowError({ code, stdout, stderr }) {
  const message = `Command failed with code ${code}
                   stdout: ${stdout}
                   stderr: ${stderr}`;
  console.error(message); // Print to console
  cy.log(message); // Log to Cypress
  throw new Error(message);
}
```

Reply:
Fede made a suggestion here to split the methods in two, I'm going to implement that so there will be two separate commands for Get and Verify.

Reply:
ack, but if you see that you need to use many times these prints, I would create common method for:
```
console.error(message);
cy.log(message); 
throw new Error(message);
```

Reply:
I've made the changes suggested by Fede and will commit them shortly. Note that the other methods in the 'projects.ts' file do not use a common function for the error handling.

Would you like me to add this to the current PR or could we refactor this later? If I add it to the current PR, should all the other methods in the 'projects.ts' file also be refactored for consistency?

Reply:
It would be nice to use a common method. If it's not an issue to add it to this PR it would be nice.
We probably need a new PR only for the other infra improvement using `cy.step`, since it imports an external plugin.

Reply:
https://github.com/opendatahub-io/odh-dashboard/pull/3388/files#diff-0800fa4ff711a8478b16382cabb7407275a1245800c19c72970c196197731573